source/grid/grid_generator.cc

```diff
@@ -3111,7 +3112,7 @@ namespace GridGenerator
                   = cell->face(i);
 
                 const Point<3> face_center (face->center());
-                if (std::abs((face_center(0)-center(0)) > 1.e-3))
+                if (std::abs(face_center(0)-center(0)) > 1.e-6)
                   {
```

Comment:
I'm all for it, but I suspect that in many cases what the tolerance should be really depends on the context. In the FE classes, there are many places where the tolerance is relative to something we have hand-tuned over the years and that depends on the polynomial degree. (Higher degree -> higher tolerance.)